### PR TITLE
Remove afna false-positive

### PR DESCRIPTION
**Pull request by Maijin**
Remove afna false-positive

### CODE REVIEW

File: libr/core/anal.c [hunk 1]
```diff
@@ -255,6 +255,16 @@ R_API char *r_core_anal_fcn_autoname(RCore *core, ut64 addr, int dump) {
 				if (dump) {
 					r_cons_printf ("0x%08"PFMT64x" 0x%08"PFMT64x" %s\n", ref->at, ref->addr, f->name);
 				}
+				if (strstr (f->name, "__stack_chk_guard"))
```
Comment (radare) [inline libr/core/anal.c]: put this logic in a separate function, with an array and a loop.
if (blacklistedWord(f->name)) { break; }

File: libr/core/anal.c [hunk 2]
```diff
@@ -255,6 +255,16 @@ R_API char *r_core_anal_fcn_autoname(RCore *core, ut64 addr, int dump) {
...
+					break;
+				if (strstr (f->name, "__stderrp"))
```
Comment (radare) [inline libr/core/anal.c]: use braces

**Comment by Maijin** [comment]
Hum I'm assuming that there is a util/ something already doing this check against a list no?

File: libr/core/anal.c [hunk 3]
```diff
@@ -239,6 +239,17 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
 		}
 	}
 }
+
+static bool blacklisted_word(char* name) {
+	const char * list[] = {"__stack_chk_guard", "__stderrp", "__stdinp", "__stdoutp", 
+	"_DefaultRuneLocale"};
+	int i;
+	for (i = 0; i < sizeof(list)/sizeof(list[0]); i++) {
+        	if (strstr(name, list[i]) != NULL) return true;
```
Comment (radare) [inline libr/core/anal.c]: add braces, remove the unnecessary != NULL and add spaces before ( and between /, etc

File: libr/core/anal.c [hunk 4]
```diff
@@ -239,6 +239,17 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
 		}
 	}
 }
+
+static bool blacklisted_word(char* name) {
+	const char * list[] = {"__stack_chk_guard", "__stderrp", "__stdinp", "__stdoutp",
```
Comment (radare) [inline libr/core/anal.c]: uh, wrongly indented the second line, missing spaces after { and before }. ending arrays with NULL is usually desirable, but i guess its fine to iterate on that constantly computed thing.

File: libr/core/anal.c [hunk 5]
```diff
@@ -255,6 +266,7 @@ R_API char *r_core_anal_fcn_autoname(RCore *core, ut64 addr, int dump) {
 				if (dump) {
 					r_cons_printf ("0x%08"PFMT64x" 0x%08"PFMT64x" %s\n", ref->at, ref->addr, f->name);
 				}
+				if (blacklisted_word(f->name)) { break; }
```
Comment (radare) [inline libr/core/anal.c]: missing space before ( and put break in the next line like this:
```
if (blacklisted_word (f->name)) {
    break;
}
```

File: libr/core/anal.c [hunk 6]
```diff
@@ -239,6 +239,16 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
 		}
 	}
 }
+
+static bool blacklisted_word(char* name) {
+	const char * list[] = {"__stack_chk_guard", "__stderrp", "__stdinp", "__stdoutp", "_DefaultRuneLocale"};
+	int i;
+	for (i = 0; i < sizeof (list) / sizeof (list[0]); i++) {
+        	if (strstr (name, list[i])) { return true; }
```
Comment (radare) [inline libr/core/anal.c]: dont be afraid of the newlines

File: libr/core/anal.c [hunk 7]
```diff
@@ -239,6 +239,16 @@ R_API void r_core_anal_autoname_all_fcns(RCore *core) {
 		}
 	}
 }
+
+static bool blacklisted_word(char* name) {
+	const char * list[] = {"__stack_chk_guard", "__stderrp", "__stdinp", "__stdoutp", "_DefaultRuneLocale"};
```
Comment (radare) [inline libr/core/anal.c]: missing space after { and before }. putting this in 3 lines will be more readable and honor the 78 column rule:
```
const char * list[] = {
      "__stack_chk_guard", "__stderrp", "__stdinp", "__stdoutp", "_DefaultRuneLocale"
};

```

**Pull request merged**